docs/source/reference/kubernetes/kubernetes-setup.rst

```diff
@@ -13,195 +13,140 @@ Kubernetes Cluster Setup
     explains how to submit tasks to your cluster.
 
 
-SkyPilot's Kubernetes support is designed to work with most Kubernetes distributions and deployment environments.
+To prepare a Kubernetes cluster to run SkyPilot, the cluster administrator must:
 
-To connect to a Kubernetes cluster, SkyPilot needs:
+1. :ref:`Deploy a cluster <kubernetes-setup-deploy>` running Kubernetes v1.20 or later.
+2. Set up :ref:`GPU support <kubernetes-setup-gpusupport>`.
+3. [Optional] :ref:`Set up ports <kubernetes-setup-ports>` for exposing services.
+4. [Optional] :ref:`Set up permissions <kubernetes-setup-serviceaccount>`: Create a namespace for your users and/or create a service account with minimal permissions for SkyPilot.
 
-* An existing Kubernetes cluster running Kubernetes v1.20 or later.
-* A `Kubeconfig <kubeconfig>`_ file containing access credentials and namespace to be used. To reduce the permissions for a user, check :ref:`required permissions guide<cloud-permissions-kubernetes>`.
+After these steps, the administrator can share the kubeconfig file with users, who can then submit tasks to the cluster using SkyPilot.
```

Comment:
I would prefer to mention the last three sections `verifying setup`, `observability`, and `troubleshooting` at the beginning as well. Otherwise, they are quite buried below.

Reply:
Good idea - I've added a toc at the top. Not including troubleshooting since it is mentioned on the index page and appears on both left and right sidebars. wdyt?